app/models/concerns/date_produced.rb

```diff
@@ -0,0 +1,30 @@
+module DateProduced
+  extend ActiveSupport::Concern
+
+  included do
+    validate :date_produced_is_a_date
+
+    def date_produced_is_a_date
```

Comment:
Although I think it doesn't make any difference in the end result, instance methods should be declare outside `included` block